### PR TITLE
Fix kernel serialization problem cause by CUPTI_ACTIVITY_KIND_KERNEL;…

### PR DESCRIPTION
**Pull request by Jokeren**
… Using concurrent kernel monitoring brings overhead for GPU kernels with many short-lived blocks. Users can use SERIALIZATION control knob to switch between the concurrent and the serialization measurement modes, depending on the number of streams used by each application

### CODE REVIEW

**Comment by Jokeren** [comment]
@dejangrubisic , please use this branch to profile PeleC, Quicksilver, and Laghos, reporting their corresponding kernel time under the two different modes.

These examples can be quickly built with the scripts here:

https://github.com/HPCToolkit/hpctoolkit-tutorial-examples/tree/nersc/examples/gpu

**Comment by Jokeren** [comment]
You can also check the differences between two modes in the trace view.

File: src/lib/prof/NameMappings.cpp [hunk 1]
```diff
@@ -47,6 +47,7 @@ const char *GPU_COPYIN  = "<gpu copyin>";
 const char *GPU_COPYOUT = "<gpu copyout>";
 const char *GPU_ALLOC   = "<gpu alloc>";
 const char *GPU_DELETE  = "<gpu delete>";
+const char *GPU_MEMSET  = "<gpu memset>";
```
Comment (Jokeren) [inline src/lib/prof/NameMappings.cpp:50]: @jmellorcrummey This is a top priority fix. Without adding the missing tag, hpcrun generates some weird call paths on summit. Then, hpcprof is confused and failed to identify the GPU kernel launch site. As a result, no GPU calling context tree will be reconstructed. 

This fix solves the above problem.

Comment (Jokeren) [reply src/lib/prof/NameMappings.cpp:50]: Update...

This does not seem to be the only culprit. There are other problems I am still investigating.

**Comment by dejangrubisic** [comment]
@jmellorcrummey @Jokeren here are results for Laghos, PeleC and QuickSilver for -ck HPCRUN_CUDA_KERNEL_SERIALIZATION

https://docs.google.com/spreadsheets/d/1hGR6-Ozy_Abr-7CUXpKoYAB2LtfDIIgN2fp4VKDyJJ8/edit?usp=sharing

1st - hpcrun -e cycles -e gpu=nvidia -t
2nd - hpcrun -ck HPCRUN_CUDA_KERNEL_SERIALIZATION=TRUE -e cycles -e gpu=nvidia -t 

Laghos        0m7.578s     | 0m6.778s
PeleC           1m13.153s   | 1m22.223s
QuickSilver  0m3.627s     | 0m3.758s

**Comment by Jokeren** [comment]
I checked the result with Dejan, and it looks reasonable to me.

In the PeleC case, the concurrent kernel mode has less overhead than the serialized mode because PeleC uses five GPU streams per process.

For the single stream Laghos, because the concurrent kernel mode instrument many small GPU kernels, the serialized mode without instrumentation can have less overhead.

**Review by jmellorcrummey — APPROVED** [review]
LGTM

**Pull request merged**